lib/supavisor/client_handler.ex

```diff
@@ -327,34 +338,36 @@ defmodule Supavisor.ClientHandler do
   end
 
   def handle_event(:timeout, :wait_ps, _, data) do
-    Logger.error("Wait parameter status timeout, send default #{inspect(data.ps)}}")
+    Logger.error(
+      "ClientHanlder: Wait parameter status timeout, send default #{inspect(data.ps)}}"
+    )
 
     ps = Server.encode_parameter_status(data.ps)
     {:keep_state_and_data, {:next_event, :internal, {:greetings, ps}}}
   end
 
   def handle_event(:timeout, :idle_terminate, _, data) do
-    Logger.warning("Terminate an idle connection by #{data.idle_timeout} timeout")
-    {:stop, :normal, data}
+    Logger.warning("ClientHanlder: Terminate an idle connection by #{data.idle_timeout} timeout")
+    {:stop, {:shutdown, :idle_terminate}}
   end
 
   def handle_event(:timeout, :heartbeat_check, _, data) do
-    Logger.debug("Send heartbeat to client")
+    Logger.debug("ClientHanlder: Send heartbeat to client")
     HH.sock_send(data.sock, Server.application_name())
     {:keep_state_and_data, {:timeout, data.heartbeat_interval, :heartbeat_check}}
   end
 
   # handle Terminate message
   def handle_event(:info, {proto, _, <<?X, 4::32>>}, :idle, _)
       when proto in [:tcp, :ssl] do
-    Logger.debug("Receive termination")
-    {:stop, :normal}
+    Logger.debug("ClientHanlder: Receive termination")
```

Comment:
This means the client close the connection yeah? 

Can we say something like "Received connection close message"

Reply:
A client sent the termination query, and then we stopped the handler

Reply:
The point is someone should be able to read it and know "client is asking us to close this connection"

"Receive termination" doesn't really express that.